packages/api/src/index.js

```diff
@@ -1,14 +1,24 @@
 /* eslint-env serviceworker */
 import { Router } from 'itty-router'
+import { HTTPError } from './errors.js'
 import { addCorsHeaders, withCorsHeaders, corsOptions } from './cors.js'
 import { envAll } from './env.js'
 import { statusGet } from './status.js'
 import { carHead, carGet, carPut, carPost } from './car.js'
 import { userLoginPost, userTokensPost, userTokensGet, userTokensDelete, userUploadsGet, userUploadsDelete, withAuth } from './user.js'
 import { metricsGet } from './metrics.js'
-import { JSONResponse, notFound } from './utils/json-response.js'
+import { notFound } from './utils/json-response.js'
 
-const router = Router()
+// Extended itty router for custom error handler
+const ThrowableRouter = (options = {}) =>
+  new Proxy(Router(options), {
+    get: (obj, prop) => (...args) =>
+      prop === 'handle'
+        ? obj[prop](...args).catch((err) => serverError(err, ...args))
+        : obj[prop](...args)
+  })
```

Comment:
😱 why do we need this?

Reply:
// Extended itty router for custom error handler

We need to be able to give `env` to the serverError function below, so that we can use `env.sentry` when catching errors from handlers.

Also, this is based on their recommendations for error handling: https://github.com/kwhitley/itty-router#extending-itty-router